numba/tests/test_parfors.py

```diff
@@ -1163,6 +1174,17 @@ def test_impl(n):
         # as a different name
         self.assertEqual(countArrayAllocs(test_impl, (types.intp,)), 1)
 
+    @skip_unsupported
+    def test_one_d_array_reduction(self):
```

Comment:
Given the scale and complexity of the changes proposed, I'm of the few that we ought to have a load more tests to try and exercise some of more of the code paths. I can help write these if desired, let me know? Thanks.

Reply:
Well, the trigger for this code is pretty precise, namely an array reduction variable.  So, I'm hopeful this won't affect non-reduction array paths.  Would be good if you could create some additional tests so that any confirmation bias I might have doesn't seep in.